backend/src/v2/driver/driver.go

```diff
@@ -357,11 +361,12 @@ func Container(ctx context.Context, opts Options, mlmd *metadata.Client, cacheCl
 	}
 
 	// Use cache and skip launcher if all contions met:
-	// (1) Cache is enabled
-	// (2) CachedMLMDExecutionID is non-empty, which means a cache entry exists
+	// (1) Cache is enabled globally
+	// (2) Cache is enabled for the task
+	// (3) CachedMLMDExecutionID is non-empty, which means a cache entry exists
 	cached := false
 	execution.Cached = &cached
-	if opts.Task.GetCachingOptions().GetEnableCache() && ecfg.CachedMLMDExecutionID != "" {
+	if *opts.CacheEnabled && opts.Task.GetCachingOptions().GetEnableCache() && ecfg.CachedMLMDExecutionID != "" {
```

Comment:
consider adding logging when opts.CacheEnabled is false so that experienced KFP users are confused by the behavior change.

Reply:
Actually, it's not a behavior change. The option defaults to `true,` and operators can set it to `false` in https://github.com/hbelmiro/data-science-pipelines/blob/d4957a32dd82e7b14be09c7607c1b59dc0fad68e/backend/src/apiserver/config/config.json#L24.
Do you think logging is needed anyway?

Reply:
I just saw [your comment on the issue](https://github.com/kubeflow/pipelines/issues/11840#issuecomment-2821021665). Yeah, it makes sense to log a message.
> Using an optional env var to override that at the deployment level is a fine option if an admin really knows what they are doing, but I bet it will catch experienced KFP users by surprise.

Reply:
"not using cache even though requested in the pipeline because it is globally disabled" or something to that effect

Reply:
thanks